### PR TITLE
zh-cn: update the translation of strings in JavaScript

### LINKED ISSUE #16820: “转义字符串中的字符”这部分最后的一个拓展链接“转义符号”链接到的内容不对
### MDN URL

https://developer.mozilla.org/zh-CN/docs/Learn/JavaScript/First_steps/Strings

### 与此议题相关的小节或标题？

“转义字符串中的字符”

### 哪些信息是错误、无用或不完善的？

“转义字符串中的字符”这部分最后的一个拓展链接“转义符号”链接到的内容不对，是链接到了“字符串String”。

### 你期望看到什么？

同步英文文档翻译，移除过时/错误内容。

### 你有什么支持的链接、参考或引用？

_No response_

### 你还想补充什么内容？

_No response_

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `zh-cn/learn/javascript/first_steps/strings`
* MDN URL: https://developer.mozilla.org/zh-CN/docs/Learn/JavaScript/First_steps/Strings
* GitHub URL: https://github.com/mdn/translated-content/blob/main/files/zh-cn/learn/javascript/first_steps/strings/index.md
* Last commit: https://github.com/mdn/translated-content/commit/3bd22c6a65f9a77e4cdd95ed4825330df9ddd710
* Document last modified: 2023-09-19T06:05:04.000Z

</details>

### PR DESCRIPTION
**Pull request by yin1999**
### Description

update the translation of strings in JavaScript

Fixes: #16820

### CODE REVIEW

File: files/zh-cn/learn/javascript/first_steps/strings/index.md [hunk 1]
```diff
@@ -17,296 +19,243 @@ slug: Learn/JavaScript/First_steps/Strings
...
+你可以使用相同的技术来插入其他特殊字符。有关更多详细信息，请参阅[转义序列](/zh-CN/docs/Web/JavaScript/Reference/Lexical_grammar#转义序列)。
+
+## 数字与字符串
+
+当我们尝试连接一个字符串和数字时会发生什么？让我们在控制台中尝试一下：
+
+```js
+const name = "Front ";
+const number = 242;
+console.log(name + number);
```
Comment (yin1999) [inline files/zh-cn/learn/javascript/first_steps/strings/index.md:230]:
```suggestion
console.log(name + number); // "Front 242"
```

**Review by jasonren0403 — COMMENTED** [review]
一些建议

File: files/zh-cn/learn/javascript/first_steps/strings/index.md [hunk 2]
```diff
@@ -1,13 +1,15 @@
 ---
-title: 文本处理 — JavaScript 中的字符串
+title: 文本处理——JavaScript 中的字符串
 slug: Learn/JavaScript/First_steps/Strings
+l10n:
+  sourceCommit: f3b6afa6ec60508dceeec4fd299ec4ba3995c0c2
```
Comment (yin1999) [inline files/zh-cn/learn/javascript/first_steps/strings/index.md:5]:
```suggestion
  sourceCommit: 3e77d220b829f7477da24af0fa4acf1a54f5dbc9
```

**Review by jasonren0403 — APPROVED** [review]
LGTM 👍

**Pull request merged**